test/workers/delete_account_hierarchy_worker_test.rb

```diff
@@ -43,19 +43,19 @@ def setup
     DeletePaymentSettingHierarchyWorker.expects(:perform_later).with(provider.payment_gateway_setting, anything, 'destroy')
     DeleteObjectHierarchyWorker.expects(:perform_later).with(api_docs_service, anything, 'destroy')
 
-    perform_enqueued_jobs { DeleteAccountHierarchyWorker.perform_now(provider) }
+    perform_enqueued_jobs(only: DeleteAccountHierarchyWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }
   end
 
   test 'does not perform if wrong state' do
     provider.update_column(:state, 'approved')
 
-    perform_enqueued_jobs { DeleteAccountHierarchyWorker.perform_now(provider) }
+    perform_enqueued_jobs(only: DeleteAccountHierarchyWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }
 
     assert provider.reload
   end
 
   test 'the account ends up destroyed after the hierarchy' do
-    perform_enqueued_jobs { DeleteAccountHierarchyWorker.perform_now(provider) }
+    perform_enqueued_jobs(except: SphinxIndexationWorker) { DeleteAccountHierarchyWorker.perform_now(provider) }
```

Comment:
This and the one above must run the same way, otherwise they are not testing the same thing.
https://github.com/3scale/porta/pull/1964/files#r440077985
IMO, this one is correct and the test above is wrong right now 😄